dataplane/sdx/sdx.go

```diff
@@ -237,10 +237,13 @@ func setupCloudStorageIfNeeded(cloudStorageBaseLocation string, instanceProfile
 func setupAwsSpotParameters(spotPercentage *int32, spotMaxPrice *float64, sdxAws **sdxModel.SdxAwsRequest) {
 	if spotPercentage != nil || spotMaxPrice != nil {
 		aws := &sdxModel.SdxAwsRequest{
-			Spot: &sdxModel.SdxAwsSpotParameters{
-				Percentage: spotPercentage,
-				MaxPrice:   *spotMaxPrice,
-			},
+			Spot: &sdxModel.SdxAwsSpotParameters{},
+		}
+		if spotPercentage != nil {
+			aws.Spot.Percentage = spotPercentage
```

Comment:
Do you know why MaxPrice and Percentage are different? MaxPrice is value the other one is a pointer...

Reply:
Go-Swagger generated the two params differently as you can see here: https://github.com/hortonworks/cb-cli/pull/753/files#diff-341f1b8b580c8da8ec569e9284c72bbf